test/integration/sa_oauthclient_test.go

```diff
@@ -78,31 +79,52 @@ func TestSAAsOAuthClient(t *testing.T) {
 	if err != nil {
 		t.Fatalf("unexpected error: %v", err)
 	}
-	if defaultSA.Annotations == nil {
-		defaultSA.Annotations = map[string]string{}
-	}
-	defaultSA.Annotations[saoauth.OAuthRedirectURISecretAnnotationPrefix+"one"] = oauthServer.URL
-	defaultSA.Annotations[saoauth.OAuthWantChallengesAnnotationPrefix] = "true"
-	defaultSA, err = clusterAdminKubeClient.ServiceAccounts(projectName).Update(defaultSA)
-	if err != nil {
-		t.Fatalf("unexpected error: %v", err)
-	}
 
-	allSecrets, err := clusterAdminKubeClient.Secrets(projectName).List(kapi.ListOptions{})
+	// retry this a couple times.  We seem to be flaking on update conflicts and missing secrets all together
+	err = wait.PollImmediate(30*time.Millisecond, 10*time.Second, func() (done bool, err error) {
```

Comment:
I think that I'm racing with one of the SA controllers, probably the one for docker configs.  It would be adding a new secret to the SA and making this fail.